### PR TITLE
Fix: Format for comprehension expressions

### LINKED ISSUE #3912: Formater should have a continuation indent
### Dafny version

4.0

### Code to produce this issue

```dafny
const x :=
  set
    i <-
      PartA +
      PartB +
      PartC,
    j <-
      PartD
    ::
      f(i,j)

const x :=
  set i <-
    PartA +
    PartB +
    PartC,
      j <-
    PartD
    ::
      f(i,j)
```

### What happened?

The indentation of the collection set is lost:

```dafny
const x :=
  set
    i <-
    PartA +
    PartB +
    PartC,
    j <-
    PartD
    ::
      f(i,j)

const x :=
  set i <-
    PartA +
    PartB +
    PartC,
      j <-
      PartD
    ::
      f(i,j)
```

Note the continuation indentation:
```dafny
set
  someTotalSet <-
    PartA +
    PartB +
    PartC
  ::
```

it may be that the standard would be to indent to the `<-`.
That is fine too, provided if I moved `<-` down it would get indented correctly.
(it does not)

### What type of operating system are you experiencing the problem on?

Mac

### PR DESCRIPTION
**Pull request by MikaelMayer**
This PR fixes #3912
Specifically, I ensured that "in" and "<-" operator indent their second operator compared to the first, but if "<-" starts the line, then it indents its second argument only relative to itself.

<small>By submitting this pull request, I confirm that my contribution is made under the terms of the [MIT license](https://github.com/dafny-lang/dafny/blob/master/LICENSE.txt).</small>

### CODE REVIEW

File: Source/DafnyCore/AST/Expressions/Expressions.cs [hunk 1]
```diff
@@ -2258,7 +2259,10 @@ public BinaryExpr(IToken tok, BinaryExpr.ResolvedOpcode rop, Expression e0, Expr
           formatter.SetIndentations(ownedTokens[0], formatter.binOpIndent, formatter.binOpIndent, formatter.binOpArgIndent);
         } else {
           var startToken = this.StartToken;
-          var newIndent = formatter.GetNewTokenVisualIndent(startToken, formatter.GetIndentInlineOrAbove(startToken));
+          var newIndent =
+            ownedTokens[0].val == "," ?
```
Comment (MikaelMayer) [inline Source/DafnyCore/AST/Expressions/Expressions.cs:2263]: "," in a comprehension is an "&&", except that it should not try to keep a visual indentation between components.

File: Source/DafnyPipeline.Test/FormatterIssues.cs [hunk 2]
```diff
@@ -4,6 +4,83 @@ namespace DafnyPipeline.Test;
...
+    FormatterWorksFor(@"
+const sameLineOp :=
+  set i
+    <-
+      PartA +
+      PartB +
+      PartC,
+    j
+    <- PartD +
+       PartE
```
Comment (alex-chew) [inline Source/DafnyPipeline.Test/FormatterIssues.cs:78]: Is this `PartE` intended to be aligned with `PartD`, or is it supposed to be one 2-space indent past the corresponding `<-`?

**Review by alex-chew — APPROVED** [review]
LGTM, aside from a question about the resulting formatting (not blocking)

File: Source/DafnyPipeline.Test/FormatterIssues.cs (hunk 2 above)
Comment (MikaelMayer) [reply Source/DafnyPipeline.Test/FormatterIssues.cs:78]: It is intended to be aligned with PartD. Specifically, the "<-" is treated like an "in" binary operator, and I put the rule that if the "in" operator is not followed by a newline, then it starts [a new aligning indentation](https://github.com/dafny-lang/dafny/pull/3916/files#diff-99ac3ad316fe9cac2a05db0a8db42d2b1fe665fb21403fb25b02e50916005c3fR2350). If it is followed by a newline, then [it's indented by two spaces as usual](https://github.com/dafny-lang/dafny/pull/3916/files#diff-99ac3ad316fe9cac2a05db0a8db42d2b1fe665fb21403fb25b02e50916005c3fR2348). This matches usages of newlines I've seen for the second argument of "in" and "<-"

**Review by alex-chew — APPROVED** [review]

**Review by alex-chew — DISMISSED** [review]

**Review by alex-chew — APPROVED** [review]

**Pull request merged**